src/OrchardCore.Modules/OrchardCore.AdminDashboard/Services/DashboardPartContentTypeDefinitionHandler.cs

```diff
@@ -0,0 +1,100 @@
+    /// This occurs during the content type building process, allowing the content type to function as a dashboard widget.
+    /// </summary>
+    public void ContentTypeBuilding(BuildingContentTypeContext context)
+    {
+        if (!context.Record.Settings.TryGetPropertyValue(nameof(ContentTypeSettings), out var node))
+        {
+            return;
+        }
+
+        var settings = node.ToObject<ContentTypeSettings>();
+
+        if (settings.Stereotype == null || !string.Equals(settings.Stereotype, AdminDashboardConstants.Stereotype, StringComparison.OrdinalIgnoreCase))
+        {
+            return;
+        }
+
+        if (context.Record.ContentTypePartDefinitionRecords.Any(x => x.Name.EqualsOrdinalIgnoreCase(nameof(DashboardPart))))
+        {
+            return;
+        }
+
+        context.Record.ContentTypePartDefinitionRecords.Add(new ContentTypePartDefinitionRecord
+        {
+            Name = nameof(DashboardPart),
+            PartName = nameof(DashboardPart),
+            Settings = new JsonObject()
+            {
+                [nameof(ContentSettings)] = JObject.FromObject(new ContentSettings
+                {
+                    IsSystemType = true,
+                }),
+            },
+        });
+    }
+
+    /// <summary>
+    /// Marks the part on the content type as a system type to prevent its removal.
+    /// This ensures that the part remains integral to the content type and cannot be deleted.
+    /// </summary>
+    public void ContentTypePartBuilding(ContentTypePartContextBuilding context)
+    {
+        if (!context.Record.PartName.EqualsOrdinalIgnoreCase(nameof(DashboardPart)))
+        {
+            return;
+        }
+
+        var settings = context.Record.Settings[nameof(ContentSettings)]?.ToObject<ContentSettings>()
+            ?? new ContentSettings();
+
+        settings.IsSystemType = true;
+
+        context.Record.Settings[nameof(ContentSettings)] = JObject.FromObject(settings);
+    }
+
+    /// <summary>
+    /// Creates a definition if the Record is null and the part name is 'DashboardPart'.
+    /// This ensures that the 'DashboardPart' has a valid definition when it is missing.
+    /// </summary>
+    public void ContentPartDefinitionBuilding(ContentPartDefinitionContextBuilding context)
+    {
+        if (context.Record is not null || context.Name != nameof(DashboardPart))
+        {
+            return;
+        }
+
+        context.Record = new ContentPartDefinitionRecord()
+        {
+            Name = context.Name,
+            Settings = new JsonObject()
+            {
+                [nameof(ContentPartSettings)] = JObject.FromObject(new ContentPartSettings
+                {
+                    Attachable = false,
+                    Reusable = true,
+                }),
+                [nameof(ContentSettings)] = JObject.FromObject(new ContentSettings
+                {
+                    IsSystemType = true,
```

Comment:
I have the feeling this should be a property in ContentPartSettings. And if a type needs a similar property let it be too. It seems nice to create a settings class because we share the same property, but it doesn't save much (space? chars?) and we can't create a class for each property that is common to two concepts. The asshole argument: Do we need a reusable class every time something has a `Name`? We don't, we just add the property where need be.

Reply:
Should the one in parts be `IsSystemPart` ?

Reply:
We'll the idea behind ContentSettings is that it can be for type, part or fields. Which is why originally it was called IsCodeManaged. Should we rename it back to IsCodeManaged or you have a better name?

Reply:
I am renaming it to `IsRemovable` for now.

Reply:
@sebastienros its probably better naming this property to somewhere where the default value is false like IsCodeManaged or IsSystemManaged or IsSystemDefined so that it would explicitly set to true.